app/src/server/api/routers/applications.ts

```diff
@@ -134,18 +135,80 @@ export const applicationsRouter = createTRPCRouter({
             },
             with: { village: { columns: { name: true } } },
           },
+          approvals: {
+            with: { approver: { columns: { userId: true, username: true, avatar: true } } },
+          },
         },
         orderBy: [desc(staffApplication.createdAt)],
         limit,
         offset: skip,
       });
 
-      // Client-side filter on username if needed (drizzle query builder with relations
-      // does not support like on related fields directly without a manual join)
+      // Attach current user's vote (if any) to each application
+      const enriched = results.map((app) => {
+        const myApproval = Array.isArray(app.approvals)
+          ? app.approvals.find((a) => a.approverUserId === user.userId)
+          : undefined;
+        return {
+          ...app,
+          myVote: myApproval?.state ?? null,
+        };
+      });
+
       const nextCursor = results.length < limit ? null : currentCursor + 1;
-      return { data: results, nextCursor };
+      return { data: enriched, nextCursor };
     }),
 
+  delete: protectedProcedure
+    .input(z.object({ id: z.string() }))
+    .output(baseServerResponse)
+    .mutation(async ({ ctx, input }) => {
+      const user = await fetchUser(ctx.drizzle, ctx.userId);
+      if (!user) return errorResponse("Not allowed");
+      if (!canDeleteStaffApplication(user.role)) return errorResponse("Not allowed");
+
+      const app = await ctx.drizzle.query.staffApplication.findFirst({
```

Comment:
user & app can be fetched in parallel at the store of the mutation for optimization. We want each endpoint to be as efficient as possible, as that reduce our server cost.